src/Models/HydrostaticFreeSurfaceModels/calculate_hydrostatic_free_surface_boundary_tendencies.jl

```diff
@@ -0,0 +1,57 @@
+import Oceananigans.Distributed: compute_boundary_tendencies!
+using Oceananigans.Utils: worktuple, offsets
+using Oceananigans.TurbulenceClosures: required_halo_size
+using Oceananigans.Models.NonhydrostaticModels: boundary_tendency_kernel_parameters,
+                                                boundary_p_kernel_parameters, 
+                                                boundary_κ_kernel_parameters,
+                                                boundary_parameters
+
+import Oceananigans.Distributed: compute_boundary_tendencies!
+
+# We assume here that top/bottom BC are always synched (no partitioning in z)
+function compute_boundary_tendencies!(model::HydrostaticFreeSurfaceModel)
+    grid = model.grid
+    arch = architecture(grid)
+
+    # We need new values for `w`, `p` and `κ`
+    recompute_auxiliaries!(model, grid, arch)
+
+    # parameters for communicating North / South / East / West side
+    kernel_parameters = boundary_tendency_kernel_parameters(grid, arch)
+    calculate_hydrostatic_free_surface_tendency_contributions!(model, kernel_parameters)
+
+    return nothing
+end
+
+function recompute_auxiliaries!(model::HydrostaticFreeSurfaceModel, grid, arch)
+    
+    w_kernel_parameters = boundary_w_kernel_parameters(grid, arch)
+    p_kernel_parameters = boundary_p_kernel_parameters(grid, arch)
+    κ_kernel_parameters = boundary_κ_kernel_parameters(grid, model.closure, arch)
+
+    for (wpar, ppar, κpar) in zip(w_kernel_parameters, p_kernel_parameters, κ_kernel_parameters)
+        compute_w_from_continuity!(model.velocities, arch, grid; parameters = wpar)
+        update_hydrostatic_pressure!(model.pressure.pHY′, arch, grid, model.buoyancy, model.tracers; parameters = ppar)
+        calculate_diffusivities!(model.diffusivity_fields, model.closure, model; parameters = κpar)
+    end
```

Comment:
Should all of this be in `update_state!`? Somehow all of this functionality is quite specific to the current form of update state; ie if we want to change how we conceive of the auxiliary state of the hydrostati cmodel, we need to change the stuff here _and_ the stuff in update state.

If we are going to create these algorithm dependencies we should write down / document what the algorithm is I think.

Reply:
you mean maybe writing an `update_auxiliaries!(model::HydrostaticFreeSurfaceModel; kwargs...)` to call both in `update_state!` and here?